src/xtgeo/surface/_regsurf_cube_window_attributes.py

```diff
@@ -0,0 +1,422 @@
+"""Regular surface vs Cube, slice a window interval v4, in pure numpy.
+
+This module is a refactored version of the original code, with the purpose of making
+it faster and more reliable.
+
+"""
+
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import numpy as np
+from scipy.interpolate import interp1d
```

Comment:
Yes I was in doubt here. The main reason is numpy's linear interpolation is only in 1D (you will have to loop over all all i, j and operate column by column, while scipy's interp1d can be vectorised to 3D).  

But I see also that vectorising all is quite memory demanding, so I am looking into a solution that operates trace by trace. This will require more testing, so it can take some time before I update this PR.